.github/workflows/ci.yaml

```diff
@@ -11,8 +11,99 @@ on:
   pull_request:
 
 jobs:
-  test:
-    name: Test Rails
-    uses: alphagov/govuk-infrastructure/.github/workflows/test-rails.yaml@main
+  cache-dependencies:
+    name: Cache dependencies
+    uses: alphagov/govuk-infrastructure/.github/workflows/cache-dependencies.yaml@main
+
+  precompile-assets:
+    name: Precompile Rails assets
+    needs: cache-dependencies
+    uses: alphagov/govuk-infrastructure/.github/workflows/precompile-rails-assets.yaml@main
+
+  security-analysis:
+    name: Security Analysis
+    needs: cache-dependencies
+    uses: alphagov/govuk-infrastructure/.github/workflows/brakeman.yaml@main
+
+  lint-scss:
+    name: Lint SCSS
+    needs: cache-dependencies
+    uses: alphagov/govuk-infrastructure/.github/workflows/stylelint.yaml@main
+    with:
+      files: "app/assets/stylesheets/"
+
+  lint-javascript:
+    name: Lint JavaScript
+    needs: cache-dependencies
+    uses: alphagov/govuk-infrastructure/.github/workflows/standardx.yaml@main
     with:
-      requiresJavaScript: true
+      files: "'app/assets/javascripts/**/*.js' 'spec/javascripts/**/*.js'"
```

Comment:
I'd wonder if you could avoid repetition by having these call the yarn task directly. e.g. `yarn lint:js`. I imagine doing that could allow you to have just one set-up yarn action and would mean you don't need to change both package.json and this if you change the JS files linted.

These shared actions have the issue that they're coupled to yarn which we don't tend to use on non-Rails projects, so might be clearer if we used a set-up yarn action (as per: https://github.com/alphagov/govuk-infrastructure/blob/main/.github/actions/setup-node/action.yaml)

I also like to dream that at some point we'll port tools to ES6 and then be able to use standard again for linting (which we do in node projects), so it wouldn't surprise me if standardx wasn't the nor

Reply:
Yah - I was trying minimise coupling this reusable workflows to dependencies within the individual repo itself. I.e. we'd need to assume that the calling repo has the "lint:js" yarn script defined. I do realise it's still coupled and assumes yarn is being used, but the reusable workflow could be iterated to support npm as well.

If we wrote this job to use the setup-node composite action as a precursor step, then we couldn't call a re-usable workflow and end up having to define the full job in every repo.

Yup - would be great to go back to standard!

Reply:
Sure, I'd have thought we could have a workflow that just allowed the task to be specified such as:

```
  lint-javascript:
    needs: cache-dependencies
    uses: alphagov/govuk-infrastructure/.github/workflows/node-task.yaml@main
    with:
      task: yarn lint:js
```